src/optimizer/query_planner.c

```diff
@@ -8622,6 +8633,48 @@ qo_search_partition_join (QO_PLANNER * planner, QO_PARTITION * partition, BITSET
 			}
 		    }
 
+		  /* check for join-connectivity of r_node to f_node */
+		  found_r_edge = found_other_edge = false;	/* init */
+		  for (t = bitset_iterate (&remaining_terms, &bt); t != -1 && !found_other_edge;
+		       t = bitset_next_member (&bt))
+		    {
+		      term = QO_ENV_TERM (env, t);
+
+		      if (!QO_IS_EDGE_TERM (term))
+			{
+			  continue;
+			}
+
+		      if (!BITSET_MEMBER (QO_TERM_NODES (term), QO_NODE_IDX (r_node)))
+			{
+			  continue;
+			}
+
+		      /* check for f_node's edges */
+		      if (BITSET_MEMBER (QO_TERM_NODES (term), QO_NODE_IDX (f_node)))
+			{
+			  /* edge between f_node and r_node */
+
+			  for (i = 0; i < QO_TERM_CAN_USE_INDEX (term) && !found_r_edge; i++)
+			    {
+			      if (QO_NODE_IDX (QO_SEG_HEAD (QO_TERM_INDEX_SEG (term, i))) == QO_NODE_IDX (r_node))
+				{
+				  found_r_edge = true;	/* indexable edge */
+				}
+			    }
+			}
+		      else
+			{
+			  /* edge between f_node and other_node */
+			  found_other_edge = true;
+			}
+		    }
+
+		  if (!found_f_edge || found_other_edge)
```

Comment:
Maybe found_r_edge should come instead of found_f_edge?

Reply:
If the newly added code is applied, the` if (cmp == PLAN_COMP_LT) block` and the `if (cmp == PLAN_COMP_GT) block` are almost identical, so extracting it as a function would be better for readability and subsequent maintenance.

Reply:
I fixed it. 
Rewriting as a function is a good idea. However, I don't want to make any further modifications to this routine which can be fully removed.